### DISCUSSION TITLE
Legacy sync mode explained

### DISCUSSION THREAD

**Discussion by lrozenblyum** [keycloak/keycloak#22335] [Q&A]
Hello!
KEYCLOAK-11862 added nice support to make synchronization of OIDC users more flexible.
However it's unclear what does the 'Legacy' mode mean

![изображение](https://github.com/keycloak/keycloak/assets/1285898/85f975a0-a251-40d6-b4ae-e93f1bf71156)

The tooltip says about keeping the behaviour before this feature was introduced. But what was the behaviour?

Was the difference related to 'when' the synchronization occurred or just to the behaviour of built-in mappers?

Thanks for clarification.

**Comment by jeetparte** [comment]
From what I noticed, 'legacy' allows custom mappers to grab the new user profile from IDP (inside the `AbstractIdentityProviderMapper.updateBrokeredUser` method), but doesn't actually update the Keycloak user (like 'force' does).

This is in contrast to 'import', which won't call the `updateBrokeredUser` method at all.

Still looking for an official answer though.
